opal/mca/btl/tcp/btl_tcp_proc.c

```diff
@@ -103,9 +96,270 @@ void mca_btl_tcp_proc_destruct(mca_btl_tcp_proc_t* tcp_proc)
     if(NULL != tcp_proc->proc_addrs) {
         free(tcp_proc->proc_addrs);
     }
+    OBJ_DESTRUCT(&tcp_proc->btl_index_to_endpoint);
     OBJ_DESTRUCT(&tcp_proc->proc_lock);
 }
 
+static inline int mca_btl_tcp_proc_is_proc_left(opal_process_name_t a,
```

Comment:
proc_left is just so that when two ranks, a and b are trying to match interfaces, the graphs created will be identical. Thus, if on rank A, the bipartite flows from A -> B, then on rank B, the bipartite flows from A -> B as well. Thus, A is the left proc. It has no significance other than allowing these two processes to create identical graphs.